src/common/valgen.cpp

```diff
@@ -172,6 +172,28 @@ bool wxGenericValidator::TransferToWindow()
             pControl->SetValue(*m_pBool) ;
             return true;
         }
+        else if (m_pInt)
+        {
+            wxRadioButton* selected = nullptr;
```

Comment:
I don't think we need this, once we find the selected button we can just call `SetValue()` on it directly and either break or return, there is no need to keep iterating further.

Reply:
I intentionally completed the loop so that I could have the wxCHECK_MSG() verify that the rest of the buttons in the group had no validator installed.  However, if you think that checking this isn't worth the extra bother, I can indeed simplify this code.

Reply:
Ah, I had missed that, sorry. I guess we could keep if you added a comment explaining what's going but it doesn't seem to be that important.

In fact, thinking more about this, such a check would be more useful in `SetValidator()` as it would point directly to the place in the program where the mistake is being made instead of triggering at a much later moment and giving you no real idea where the problem is (how would you find the offending radio button?). So what about overriding `SetWindow()` in `wxGenericValidator` and checking that the window is not the non-first radio button in the group there?